pkg/registry/abstract.go

```diff
@@ -0,0 +1,41 @@
+package registry
```

Comment:
move to `pkg/registry/abstract/abstract.go`

ensure it implements an interface that would be generic enough between any added registry in the future. e.g. having functionality such as

```golang
type RegistryAuthenticator interface {
  EnrichAndValidate() error
  GetAuthToken() (Token, error)
}
```